### DISCUSSION TITLE
如果创建组件库就是需要使用cssModule，应该怎么配置呢，

### DISCUSSION THREAD

**Discussion by Chasen-Zhang** [umijs/dumi#1733] [Q&A]
现在是本地开发文档、业务测引用组件本地开发都是ok，但是业务测打包，模块样式就是会消失，比如
```html
<span className={styles.button}>test</span> 变成了 <span>test</span>，
```
虽说官方给的是很难覆盖样式，其实实际生产中，我可以给组件提供className 或者 style 属性的，要覆盖就这样覆盖就行了。但是现在我其实是期待可以使用css module 的

https://github.com/umijs/dumi/issues/225

 fatherrc.ts 中说可以配置cssModule，发现现在好像配置变动了。可能这个 #236 也是我遇到的问题，
 
 https://github.com/umijs/dumi/issues/650

大神求助下，感激不尽。

**Comment by mortalYoung** [comment]
> 其实实际生产中，我可以给组件提供className 或者 style 属性的，要覆盖就这样覆盖就行了

你这个属于比较简单的单层级组件，可以这么做。我私以为在实际生产中，业务组件通常会比较复杂，层级比较深的话，你这个方式不行的。我建议还是不要在组件库里考虑 cssModule 了吧

**Reply by Mirrors406** [reply]
我是一定要使用 css module 的情况，我是不是用不了 dumi 了
